lib/src/builder/execute.rs

```diff
@@ -387,7 +405,12 @@ pub fn fill_eth_tx_env(tx_env: &mut TxEnv, tx: &TxEnvelope) -> Result<(), Error>
     // TODO(Brecht): use optimized recover
     match tx {
         TxEnvelope::Legacy(tx) => {
+            #[cfg(feature = "sp1-cycle-tracker")]
+            println!("cycle-tracker-start: Legacy");
             tx_env.caller = tx.recover_signer().unwrap_or_default();
+            #[cfg(feature = "sp1-cycle-tracker")]
+            println!("cycle-tracker-end: Legacy");
```

Comment:
I think all these begin/end tracker code will be cleaner if we have some helper struct similar to   `Measurement`, so we can just do something like
 ```
let measurement = CycleMeasurement::start("legacy");
// do something
measurement.stop();
```

This also (but optimistically) allows us to easily add support for different zkVMs inside `CycleMeasurement` to measure with whatever is available. Maybe we can even just directly use `Measurement` and when running inside a zkVM we track the cycles instead of measuring the time like we do right now on all platforms, but maybe that's going a bit too far in the abstraction.